python/py-pytest-mypy/Portfile

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+github.setup        dbader pytest-mypy 0.8.0 v
+revision            0
+name                py-${github.project}
+fetch.type          git
```

Comment:
I don't see why this is needed; can you not just remove this or get use `github.tarball_from archive`?

Reply:
A git clone gets around this  setuptools-scm build issue:
```
LookupError: setuptools-scm was unable to detect version for '/opt/local/var/macports/build/_opt_local_ports_python_py-pytest-mypy/py39-pytest-mypy/work/dbader-pytest-mypy-7fdc974'.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
Command failed:  cd "/opt/local/var/macports/build/_opt_local_ports_python_py-pytest-mypy/py39-pytest-mypy/work/pytest-mypy-0.8.0" && /opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9 setup.py --no-user-cfg build -j12 
Exit code: 1
Error: Failed to build py39-pytest-mypy: command execution failed
```

Is there another approach?

Reply:
if there is no obvious reason to take the stuff from GitHub, I would personally just go with PyPI for all Python packages. Sometimes though the tests are not part of the PyPI package so it could be beneficial to use GitHub - for this specific port I don't know.

Reply:
It doesn't build if you don't clone it because the build checks if it's a git clone. That's the obvious reason. Please see the error above.

Reply:
the message you copied above states to either use a git clone or PyPI tarball, at least suggesting that the latter would work fine.

Reply:
The PyPi tarball doesn't come with tests. Building from GitHub source code is preferable.

Reply:
I agree building from source is preferred.